src/fga/interfaces/query.interface.ts

```diff
@@ -0,0 +1,33 @@
+import { Warrant, PolicyContext, WarrantResponse } from './warrant.interface';
+import { PaginationOptions } from '../../common/interfaces/pagination-options.interface';
+import { GetOptions } from '../../common/interfaces';
+
+export interface QueryOptions extends PaginationOptions {
+  q: string;
+  context?: PolicyContext;
+}
+
+export interface SerializedQueryOptions extends PaginationOptions {
+  q: string;
+  context?: string;
+}
+
+export interface QueryResult {
+  resourceType: string;
+  resourceId: string;
+  relation: string;
+  warrant: Warrant;
+  isImplicit: boolean;
+  meta?: { [key: string]: any };
```

Comment:
Nit: We'll typically reach for `Record` instead of declaring an object with an index signature:

```suggestion
  meta?: Record<string, any>;
```